tests/test_decorator.py

```diff
@@ -119,6 +119,7 @@ def foo(a, b, *args, d=3, **kwargs):
     assert foo(1, 2, 3, d=4) == 'a=1, b=2, args=(3,), d=4, kwargs={}'
     assert foo(*[1, 2, 3], d=4) == 'a=1, b=2, args=(3,), d=4, kwargs={}'
     assert foo(1, 2, 3, e=10) == "a=1, b=2, args=(3,), d=3, kwargs={'e': 10}"
+    assert foo(1, 2, kwargs=4, e=5) == "a=1, b=2, args=(), d=3, kwargs={'kwargs': 4, 'e': 5}"
```

Comment:
Hi @cybojenix and thanks for the PR! Nice catch!
Two remarks:
- Could we enforce in the test that we expect the exact same behaviour than plain python.
For example by doing something like
```python
@pytest.mark.parametrize('validated', (True, False))
def test_var_args_kwargs(validated):
    def foo(a, b, *args, d=3, **kwargs):
        return f'a={a!r}, b={b!r}, args={args!r}, d={d!r}, kwargs={kwargs!r}'

    if validated:
        foo = validate_arguments(foo)

    assert ...
```
- Don't we have the same error with `args` i.e. `foo(1, 2, args=4, e=5)`, which doesn't seem to be fixed with this PR. I reckon this could be tackled at the same time ;)

Reply:
Added. You were right, `args` also gets overwritten.